src/partest/scala/tools/partest/DirectTest.scala

```diff
@@ -53,12 +53,15 @@ abstract class DirectTest {
   // a default Settings object using only extraSettings
   def settings: Settings = newSettings(tokenize(extraSettings))
   // settings factory using given args and also debug settings
-  def newSettings(args: List[String]) = (new Settings).tap { s =>
-    val allArgs = args ++ tokenize(debugSettings)
+  def newSettings(args: List[String]): Settings = newSettings().tap { s =>
+    val allArgs = debugSettings.pipe(db => if (db.isEmpty) args else args ++ tokenize(db))
     log(s"newSettings: allArgs = $allArgs")
     val (success, residual) = s.processArguments(allArgs, processAll = false)
     assert(success && residual.isEmpty, s"Bad settings [${args.mkString(",")}], residual [${residual.mkString(",")}]")
   }
+  // scaladoc has custom settings
+  def newSettings(): Settings = new Settings
```

Comment:
Call this `newSettingsBase()` or something like that? With the same name as the overload I would expect this method to be an alias for `newSettings(args)` with some default `args`; not a base method that `newSettings(args)` *must* call to create its instance of `Settings`.